ds/ds.go

```diff
@@ -124,13 +125,35 @@ func Parse(uri string) (dsQuery, error) {
 		return result, fmt.Errorf("Not an dns-sd URI")
 	}
 
-	// following dns-sd URI conventions from CUPS
-	if u.Host != "" {
-		result.Domain = u.Host
-	}
-	if u.Path != "" {
-		// remove prefix slash
-		result.Type = u.Path[1:]
+	// following dns-sd URI conventions from CUPS dnssd://instance._type._proto.domain/?query
+	if u.Hostname() != "" {
+		parts := strings.Split(u.Hostname(), ".")
```

Comment:
it would be best if this were a function that came with tests. I am bad at this kind of string foo on a good day, and I am sure this is right, but ... phew. Anytime I see indexes being manipulated this way I worry.